packages/preview/src/browser/preview-contribution.ts

```diff
@@ -115,6 +138,7 @@ export class PreviewContribution extends NavigatableWidgetOpenHandler<PreviewWid
     protected synchronizeScrollToEditor(previewWidget: PreviewWidget, editor: TextEditor): Disposable {
         return previewWidget.onDidScroll(sourceLine => {
             const line = Math.floor(sourceLine);
+            this.delayScrollSync(200); // avoid recursive scroll synchronization
```

Comment:
it tried the timeout-based sync so many times ;-) there is always an example that breaks.

![2019-06-13 17 21 53](https://user-images.githubusercontent.com/914497/59445427-f2ba8e00-8dff-11e9-8c00-9f89ffc2c939.gif)

Reply:
could you try to reproduce by scrolling faster and opening a large file?

Reply:
Thanks a lot for your advise! I'll try it tomorrow, and see what I can do.

Reply:
@AlexTugarev Codes updated. It should work much better now.

![fix-markdown-preview-2](https://user-images.githubusercontent.com/17487291/59480832-bc7b1e00-8e94-11e9-9e1e-061ac775e3b6.gif)

The synchronization may break if you switch between scrolling on the editor and the preview quickly enough. But it has to be really quick. So in general, it works pretty well to me ：).

![fix-markdown-preview-2-break](https://user-images.githubusercontent.com/17487291/59481014-3ad7c000-8e95-11e9-8cc0-2ce4969820ab.gif)

Reply:
It's a great improvement!